plantuml/LinkLoss/disconnection.puml

```diff
@@ -0,0 +1,46 @@
+@startuml
+skinparam defaultTextAlignment center
+
+|App|
+
+|Link Loss Service|
+
+|Gap|
+start
+
+|Link Loss Service|
+:disconnection;
+-> event;
+
+:get_alert_level()|
+-> level;
+
+if (alert_handler != NULL ?) then (yes)
+    if (event.getReason() == CONNECTION_TIMEOUT ?) then (yes)
+        if (level != NO_ALERT ? ) then (yes)
+            if (~ in alert ?) then (yes)
+                :set alert flag to true;
+                :on_alert_requested()|
```

Comment:
this should be in the app column

Reply:
That's __not__ true. That event handler must be called here to trigger the event in the application.   

It may be worth duplicating the call in the app column, but removing it completely from the LLS column makes no sense to me.

Reply:
OK, normally you would put it in between the roles with an arrow pointing the in the direction of the roles being acted on, like this:
```
user              linkloss service              ble                    peer
  +                      +                       +                       +
  |  set alert callback  |                       |                   user sets remote
  +--------------------->+                       |      att write    LLS alert to high
  |                      |                       +<----------------------+
  |                      |       write callback  |                       |
  |                      +<----------------------+                       |
  |               set alert to high              |                       |
  |                      |                       |                     no|tx
  |                      |                    timeout                    |
  |                      |      on disconnect    |                       |
  |                      +<----------------------+                       |
  |                      |                       |                       |
  | call alert callback  |                       |                       |
  +<---------------------+                       |                       |
  |                      |                       |                       |
  +                      |                       |                       |
```